src/dialogs/connectsettingsdialog.cpp

```diff
@@ -0,0 +1,84 @@
+#include "connectsettingsdialog.h"
+#include "ui_connectsettingsdialog.h"
+#include <QSerialPortInfo>
+#include <QMessageBox>
+
+ConnectSettingsDialog::ConnectSettingsDialog(QStringList firmwaresList, QWidget *parent) :
+    QDialog(parent),
+    ui(new Ui::ConnectSettingsDialog),
+    deviceNotifier(Solid::DeviceNotifier::instance())
+{
+    ui->setupUi(this);
+    connect(deviceNotifier, &Solid::DeviceNotifier::deviceAdded, this, &ConnectSettingsDialog::locateSerialPort);
+    connect(deviceNotifier, &Solid::DeviceNotifier::deviceRemoved, this, &ConnectSettingsDialog::locateSerialPort);
+    locateSerialPort();
+
+    connect(ui->buttonBox, &QDialogButtonBox::accepted,[=]{
+        if(ui->profileCB->currentText().isEmpty()) {
+            QMessageBox msg;
```

Comment:
why not use QMessageBox::information ? the one line its something lik e
QMessageBox::Information(parent, icon, title, message)